src/serializer/ResidualHeapVisitor.js

```diff
@@ -581,7 +581,32 @@ export class ResidualHeapVisitor {
           if (object.intrinsicName === "global") continue; // Avoid double-counting
           this.visitObjectProperty(binding);
         }
-        // TODO #990: Fix additional functions handing of ModifiedBindings
+        // Handing of ModifiedBindings
+        for (let additionalBinding of modifiedBindings.keys()) {
+          let modifiedBinding: Binding = ((additionalBinding: any): Binding);
+          let residualBinding;
+          if (modifiedBinding.isGlobal) {
+            residualBinding = this.globalBindings.get(modifiedBinding.name);
+          } else {
+            let containingEnv = modifiedBinding.environment;
+            invariant(containingEnv instanceof DeclarativeEnvironmentRecord);
+            let bindMap = this.declarativeEnvironmentRecordsBindings.get(containingEnv);
+            if (bindMap) residualBinding = bindMap.get(modifiedBinding.name);
+          }
+          // Only visit it if there is already a binding
+          if (residualBinding && modifiedBinding.value !== residualBinding.value) {
+            let newValue = modifiedBinding.value;
+            invariant(newValue);
+            this.visitValue(newValue);
+            residualBinding.modified = true;
+            invariant(
```

Comment:
Why is this a valid invariant? Could I easily write a program with two additional functions that will violate this? Log an error instead? Indicating to the user that this is a current limitation. Invariants always indicate Prepack bugs that will get the oncall woken up at night.